arangod/RestHandler/RestCursorHandler.cpp

```diff
@@ -618,10 +623,24 @@ RestStatus RestCursorHandler::generateCursorResult(rest::ResponseCode code) {
     builder.add(StaticStrings::Error, VPackValue(false));
     builder.add(StaticStrings::Code, VPackValue(static_cast<int>(code)));
     builder.close();
+    _cursor->setLastQueryBatchObject(builder);
 
     _response->setContentType(rest::ContentType::JSON);
+    TRI_IF_FAILURE("MakeConnectionErrorForRetry") { return RestStatus::FAIL; }
+
     generateResult(code, std::move(buffer), std::move(ctx));
   } else {
+    VPackBuilder builder;
+    builder.openObject();
+    builder.add(StaticStrings::Code,
+                VPackValue(static_cast<int>(
+                    GeneralResponse::responseCode(r.errorNumber()))));
+    builder.add(StaticStrings::Error, VPackValue(true));
+    builder.add("errorMessage", VPackValue(r.errorMessage()));
+    builder.add("errorNum", VPackValue(r.errorNumber()));
+    builder.close();
+    _cursor->setLastQueryBatchObject(builder);
```

Comment:
same here: we shouldn't store a full Builder, but rather a shared_ptr to a Buffer.